### PR TITLE
Git proxy prepare

### PR DESCRIPTION
**Pull request by janfjohannes**
prepares the main branch for using an external git and github proxy with authenticated CORS

### CODE REVIEW

**Comment by samuelstroschein** [comment]
@janfjohannes should i review it?

**Comment by janfjohannes** [comment]
@samuelstroschein  yes pls :)

File: inlang/source-code/website/src/pages/install/InstallationProvider.tsx [hunk 1]
```diff
@@ -140,12 +140,23 @@ async function initializeRepo(
...
 	const repo = await openRepository(repoURL, {
 		nodeishFs: createNodeishMemoryFs(),
-		corsProxy: publicEnv.PUBLIC_GIT_PROXY_PATH,
+		corsProxy: publicEnv.PUBLIC_GIT_PROXY_BASE_URL + publicEnv.PUBLIC_GIT_PROXY_PATH,
 	})
 
-	const isCollaborator = await repo.isCollaborator({
-		username: user.username,
-	})
+	const isCollaborator = await repo
```
Comment (samuelstroschein) [inline inlang/source-code/website/src/pages/install/InstallationProvider.tsx:146]:
```suggestion
	const canCommitToRepository = await repo
```

Comment (samuelstroschein) [reply inlang/source-code/website/src/pages/install/InstallationProvider.tsx:146]: We should probably rename the API in the lix client too. It is not obvious what the scope of `isCollaborator` entails. `canCommit` or `hasCommitRights` is more explicit.

cc @jannesblobel - we discussed this too as part of the telemtry

File: inlang/source-code/website/src/services/auth/src/onSignOut.ts [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
  */
 export async function onSignOut(args: { setLocalStorage: SetStoreFunction<LocalStorageSchema> }) {
 	// sign out on the server
-	await fetch("/services/auth/sign-out", { method: "POST" })
+	const hasExternalGitProxy =
+		publicEnv.PUBLIC_SERVER_BASE_URL !== publicEnv.PUBLIC_GIT_PROXY_BASE_URL
+
+	if (hasExternalGitProxy) {
+		await Promise.allSettled([
+			fetch("/services/auth/sign-out", { method: "POST" }),
```
Comment (samuelstroschein) [inline inlang/source-code/website/src/services/auth/src/onSignOut.ts:16]: is the auth sign out API route (telefunc) still required? Couldn't the code here simply set the local storage['user'] to undefined?

File: lix/source-code/client/src/http-client.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		statusCode: res.status,
+		statusMessage: res.statusText,
+		body: iter,
+		headers: headers,
+	}
+}
+
+var index = { request }
+
+export default index
```
Comment (samuelstroschein) [inline lix/source-code/client/src/http-client.js:166]: Why this double export? 

I assume you copy & pasted the code from somewhere. The double export might be on oversight when you copied the code.

**Review by samuelstroschein — APPROVED** [review]

**Comment by samuelstroschein** [comment]
@janfjohannes overall OK. some minor naming of things, nothing important

File: inlang/source-code/website/src/services/auth/src/onSignOut.ts (hunk 2 above)
Comment (janfjohannes) [reply inlang/source-code/website/src/services/auth/src/onSignOut.ts:16]: @samuelstroschein  no as the cookie needs to be deleted which is http only.

File: inlang/source-code/website/src/pages/install/InstallationProvider.tsx (hunk 1 above)
Comment (janfjohannes) [reply inlang/source-code/website/src/pages/install/InstallationProvider.tsx:146]: @samuelstroschein  everyone can commit who can read a repo, a collaborator is allowed to push. i think we should change this when naming is clearer and not part of this pr...

**Pull request merged**